### PR TITLE
add 'ordered_by' method to QuerySet class.

### PR DESCRIPTION
**Pull request by leviathan747**
this method returns a sorted set based upon the specified attribute name.
if the class has no attribute by that name, an exception is raised.

### CODE REVIEW

File: xtuml/meta.py [hunk 1]
```diff
@@ -1067,6 +1073,52 @@ def cardinality(instance_or_set):
...
+    instance set.
+    '''
+    def __call__(self, x, y):
+        for attr in self:
+            if not hasattr(x, attr):
+                raise MetaException("Class '%s' has no attribute '%s'" % (get_metaclass(x).kind, attr))
+            if not hasattr(y, attr):
+                raise MetaException("Class '%s' has no attribute '%s'" % (get_metaclass(y).kind, attr))
+            
+        return cmp( tuple(getattr(x, attr) for attr in self), tuple(getattr(y, attr) for attr in self) )
```
Comment (john-tornblom) [inline xtuml/meta.py]: superfluous spaces within parenthesis group

File: tests/test_xtuml/test_metamodel.py [hunk 2]
```diff
@@ -66,6 +68,23 @@ def test_select_any_where(self):
         m = self.metamodel
         inst = m.select_any('S_DT', where(Name='void'))
         self.assertEqual(inst.Name, 'void')
+
+    def test_select_many_ordered_by(self):
+        m = self.metamodel
+        q = m.select_many('S_DT', None, order_by('Name', 'DT_ID'))
```
Comment (john-tornblom) [inline tests/test_xtuml/test_metamodel.py:74]: I was thinking, it should be possible to to make the where_clause and order_by argument less independent on the order in the parameter list. Something like this:

def select_many(kind, set_modifiers*):
   s = map(apply_set_operator, set_modifiers)
   return s

then you could type:
    q = m.select_many('S_DT', order_by('Name', 'DT_ID'), where(...))
or
    q = m.select_many('S_DT', where(...), order_by('Name', 'DT_ID'))
or
    q = m.select_many('S_DT', order_by('Name', 'DT_ID'))

File: xtuml/meta.py [hunk 3]
```diff
@@ -1067,6 +1073,52 @@ def cardinality(instance_or_set):
...
+    set based on attribute names passed.  When ordering on multiple
+    attributes is specified, the set will be sorted by the first
+    attribute and then within each value of this, by the second
+    attribute and so on.
+    
+    Usage example:
+    
+    >>> from xtuml import order_by
+    >>> m = xtuml.load_metamodel('db.sql')
+    >>> inst = m.select_many('My_Modeled_Class', None, order_by( 'Name', 'Number' ) )
```
Comment (john-tornblom) [inline xtuml/meta.py]: Again, superfluous spaces within parenthesis group.

File: tests/test_xtuml/test_metamodel.py (hunk 2 above)
Comment (leviathan747) [reply tests/test_xtuml/test_metamodel.py:74]: hmmm. I will give this some thought. Of course the difficulty is how to implement the 'apply_set_operator' part

Comment (john-tornblom) [reply tests/test_xtuml/test_metamodel.py:74]: map() might have been a bad idea. I slapped together a gist demonstrating one way of doing it. needs some cleaning though...

See https://gist.github.com/john-tornblom/21eca04788c3980778039bf91866e7dd

File: xtuml/meta.py (hunk 1 above)
Comment (leviathan747) [reply xtuml/meta.py]: done

File: xtuml/meta.py (hunk 3 above)
Comment (leviathan747) [reply xtuml/meta.py]: done

**Comment by john-tornblom** [comment]
Python3 seems to fail on travis, see https://api.travis-ci.org/v3/job/391935957/log.txt

I was thinking of altering your patch a bit anyway to something like this: https://gist.github.com/john-tornblom/21eca04788c3980778039bf91866e7dd). I can accept the patch and rework it myself, or do you wanna give it a go?

**Comment by leviathan747** [comment]
You can either accept mine and alter as you wish or you could submit a PR to my fork and I will service it (and then this PR would get automatically updated).

**Comment by john-tornblom** [comment]
My repo was created before xtuml/pyxtuml, and cort did not fork via github so not sure how to send a PR directly. But I've pushed to https://github.com/john-tornblom/pyxtuml/tree/ordered_by

**Pull request merged**